src/lib/currency/currency.ml

```diff
@@ -727,47 +718,33 @@ end = struct
 
     let sub_flagged x y =
       let%bind z = seal (Field.Var.sub x y) in
-      let%map z, `Overflow underflow = range_adjust_flagged `Sub z in
+      let%map z, `Overflow underflow = range_check_flagged `Sub z in
       (z, `Underflow underflow)
 
     let sub_or_zero x y =
-      make_checked (fun () ->
-          let open Tick.Run in
-          let res = Pickles.Util.seal Tick.m Field.(x - y) in
-          let neg_res = Pickles.Util.seal Tick.m (Field.negate res) in
-          let x_gte_y = run_checked (range_check_flag res) in
-          let y_gte_x = run_checked (range_check_flag neg_res) in
-          Boolean.Assert.any [ x_gte_y; y_gte_x ] ;
-          (* If y_gte_x is false, then x_gte_y is true, so x >= y and
-             thus there was no underflow.
-
-             If y_gte_x is true, then y >= x, which means there was underflow
-             iff y != x.
-
-             Thus, underflow = (neg_res_good && y != x)
-          *)
-          let underflow =
-            Boolean.( &&& ) y_gte_x (Boolean.not (Field.equal x y))
-          in
-          Field.if_ underflow ~then_:Field.zero ~else_:res)
+      let%bind res, `Underflow underflow = sub_flagged x y in
+      Field.Checked.if_ underflow ~then_:Field.(Var.constant zero) ~else_:res
 
     let assert_equal x y = Field.Checked.Assert.equal x y
 
     let equal x y = Field.Checked.equal x y
 
     let ( = ) = equal
 
-    (* x <= y iff range_check_flag (y - x) *)
-    let ( <= ) x y = range_check_flag (Field.Var.sub y x)
+    let ( < ) x y =
+      let%bind diff = seal (Field.Var.sub x y) in
+      (* [lt] is true iff [x - y < 0], ie. [x < y] *)
+      let%map _res, `Overflow lt = range_check_flagged `Sub diff in
```

Comment:
Shouldn't this be `Underflow`?

Reply:
`range_check_flagged` is supposed to do a bi-directional check, or just check for overflow/underflow depending on the choice between `` `Add`` `` `Sub`` and `` `Add_or_sub``. That part of the return value could be tagged `` `Outflow` instead, if that were clearer

Reply:
Either that or a comment would work